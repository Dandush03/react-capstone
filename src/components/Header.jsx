import React from 'react';

import BarButton from './BarButton';
import Media from './Media';
import Menu from './Menu';

export default function Header() {
  return (
    <header>
      <div>
        <BarButton />
        <Media />
      </div>
      <Menu />
    </header>
  );
}
