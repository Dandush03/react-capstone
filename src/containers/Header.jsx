import React from 'react';

import BarButton from '../components/BarButton';
import Media from '../components/Media';
import Menu from '../components/Menu';

export default function Header() {
  return (
    <header>
      <div>
        <BarButton />
        <Media />
      </div>
      <Menu />
    </header>
  );
}
