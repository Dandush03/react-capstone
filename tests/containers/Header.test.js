import React from 'react';
import renderer from 'react-test-renderer';

import { Provider } from 'react-redux';
import storeConfig from '../../src/store';
import Header from '../../src/containers/Header';


const store = storeConfig();

describe('components', () => {
  describe('Header', () => {
    const component = renderer.create(<Provider store={store}><Header /></Provider>);
    const tree = component.toJSON();
    it('Should match to Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
