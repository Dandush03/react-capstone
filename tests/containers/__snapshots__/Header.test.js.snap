// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Header Should match to Snapshot 1`] = `
<header>
  <div>
    <button
      className="hamburger"
      onClick={[Function]}
      type="button"
    >
      <div
        className="bar1"
      />
      <div
        className="bar2"
      />
      <div
        className="bar3"
      />
    </button>
    <div
      className="media-icons"
    >
      <a
        className="linkedin-hero"
        href="https://www.linkedin.com/in/daniel-laloush"
      >
        <svg
          aria-hidden="true"
          className="linkedin"
          data-icon="linkedin"
          data-prefix="fab"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
            fill="currentColor"
          />
        </svg>
      </a>
      <a
        className="hero-page"
        href="https://dlaloush.me/"
      >
        <svg
          aria-hidden="true"
          className="hero"
          data-icon="paper-plane"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M476 3.2L12.5 270.6c-18.1 10.4-15.8 35.6 2.2 43.2L121 358.4l287.3-253.2c5.5-4.9 13.3 2.6 8.6 8.3L176 407v80.5c0 23.6 28.5 32.9 42.5 15.8L282 426l124.6 52.2c14.2 6 30.4-2.9 33-18.2l72-432C515 7.8 493.3-6.8 476 3.2z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
  </div>
  <menu>
    <ul>
      <li>
        <a
          href="/"
        >
          <span>
            Home
          </span>
          <svg
            aria-hidden="true"
            className="home"
            data-icon="home"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
              fill="currentColor"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          href="/compare"
        >
          <span>
            Compare
          </span>
          <svg
            aria-hidden="true"
            className="info-svg"
            data-icon="info"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 192 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v212.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 24 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z"
              fill="currentColor"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          <span>
            Company
          </span>
          <svg
            aria-hidden="true"
            className="company-svg"
            data-icon="building"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M436 480h-20V24c0-13.255-10.745-24-24-24H56C42.745 0 32 10.745 32 24v456H12c-6.627 0-12 5.373-12 12v20h448v-20c0-6.627-5.373-12-12-12zM128 76c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12V76zm0 96c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40zm52 148h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12zm76 160h-64v-84c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v84zm64-172c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40zm0-96c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40zm0-96c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12V76c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40z"
              fill="currentColor"
            />
          </svg>
        </a>
      </li>
    </ul>
  </menu>
</header>
`;
